apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/RLSManagement/RLSManagement.tsx

```diff
@@ -0,0 +1,401 @@
+import type { PostgresTable } from '@supabase/postgres-meta'
+import { useEffect, useMemo, useState } from 'react'
+
+import { AIOptInModal } from 'components/ui/AIAssistantPanel/AIOptInModal'
+import { ToggleRlsButton } from 'components/ui/ToggleRlsButton'
+import { useDatabasePoliciesQuery } from 'data/database-policies/database-policies-query'
+import { useSelectedProjectQuery } from 'hooks/misc/useSelectedProject'
+import { toast } from 'sonner'
+import type { ResponseError } from 'types'
+import { Button, Card, CardContent, cn } from 'ui'
+import { Admonition } from 'ui-patterns/admonition'
+
+import {
+  generateStartingPoliciesForTable,
+  type GeneratedPolicy,
+} from 'components/interfaces/Auth/Policies/Policies.utils'
+import { generatePolicyUpdateSQL } from 'components/interfaces/Auth/Policies/PolicyTableRow/PolicyTableRow.utils'
+import {
+  useForeignKeyConstraintsQuery,
+  type ForeignKeyConstraint,
+} from 'data/database/foreign-key-constraints-query'
+import { useOrgAiOptInLevel } from 'hooks/misc/useOrgOptedIntoAi'
+import { useTrack } from 'lib/telemetry/track'
+import { ExternalLink, InfoIcon } from 'lucide-react'
+import Link from 'next/link'
+import type { ForeignKey } from '../../ForeignKeySelector/ForeignKeySelector.types'
+import type { ColumnField } from '../../SidePanelEditor.types'
+import { PolicyList, type PolicyListItemData } from './PolicyList'
+
+interface RLSManagementProps {
+  schema: string
+  table?: PostgresTable
+  tableName?: string // For new tables
+  columns?: ColumnField[] // For new tables
+  foreignKeyRelations?: ForeignKey[] // For new tables
+  isRlsEnabled: boolean
+  onChangeRlsEnabled?: (isEnabled: boolean) => void
+  isNewRecord: boolean
+  isDuplicating: boolean
+  generatedPolicies?: GeneratedPolicy[]
+  onGeneratedPoliciesChange?: (policies: GeneratedPolicy[]) => void
+}
+
+export const RLSManagement = ({
+  schema,
+  table,
+  tableName,
+  columns = [],
+  foreignKeyRelations = [],
+  isRlsEnabled,
+  onChangeRlsEnabled,
+  isNewRecord,
+  isDuplicating,
+  generatedPolicies = [],
+  onGeneratedPoliciesChange,
+}: RLSManagementProps) => {
+  const { data: project } = useSelectedProjectQuery()
+  const [isGenerating, setIsGenerating] = useState(false)
+  const [generateFailed, setGenerateFailed] = useState(false)
+  const [isOptInModalOpen, setIsOptInModalOpen] = useState(false)
+  const { includeSchemaMetadata } = useOrgAiOptInLevel()
+  const track = useTrack()
+  const isExistingTable = !!table && !isNewRecord && !isDuplicating
+  const rlsEnabled = isRlsEnabled ?? false
```

Comment:
could declare `isRlsEnabled` to have a default value of `false` so we don't need an additional variable